### PR TITLE
#1468 Skip pointless actions if SIM algo already converged

### LINKED ISSUE #1468: Avoid useless curves update and criteria evaluation in stabilized situation with simplified solver
In the current state, when we have reached a stabilized situation (ie no further Newton done), the curves are updated at each time step as well as the log and the criteria are evaluated. 
It is not necessary in this case.

The idea would be to avoid doing this. Certainly a proper way will be to handle this in the simplified solver implementation and not at the simulation level, for example by calling again the solve method in these cases.

### PR DESCRIPTION
**Pull request by lecourtoisflo**
closes #1468

### CODE REVIEW

**Review by gautierbureau — APPROVED** [review]

File: dynawo/sources/Solvers/Common/DYNSolver.h [hunk 1]
```diff
@@ -47,7 +47,8 @@ typedef enum {
   ModeChange = 0x01,
   NotSilentZChange = 0x02,
   SilentZNotUsedInDiscreteEqChange = 0x04,
-  SilentZNotUsedInContinuousEqChange = 0x08
+  SilentZNotUsedInContinuousEqChange = 0x08,
+  NRSkipped = 0x10
 } StateFlags;
```
Comment (adriguir) [inline dynawo/sources/Solvers/Common/DYNSolver.h:52]: I think it should'nt be part of this enum as it encompasses a quite different meaning.
The enum is currently related to event handling scheme and to what is going on in terms of discrete variables and mode changes and so integrating here the NRSkipped doesn't seem the best option in my opinion.

I would definetely prefer to have a specific boolean or a specific status than reusing the existing one.

**Review by adriguir — CHANGES_REQUESTED** [review]

File: dynawo/sources/Solvers/Common/DYNSolver.h (hunk 1 above)
Comment (lecourtoisflo) [reply dynawo/sources/Solvers/Common/DYNSolver.h:52]: even if it means duplicating the mechanism ?

**Comment by adriguir** [comment]
I don't like the wording (even if similar to what was there in Simplified solver) and I'm not sure if the computation gain deserves the effort.
@gautierbureau @rosiereflo what do you think? should we do this dev finally or not?

**Comment by rosiereflo** [comment]
> I don't like the wording (even if similar to what was there in Simplified solver) and I'm not sure if the computation gain deserves the effort. @gautierbureau @rosiereflo what do you think? should we do this dev finally or not?

I agree that the wording is not good as it is really related to the simplified solver (it should be something like "modelChanged" for example). This will be efficient if we have use cases with a very large amount of criteria or curves, which is not our typical use case. Moreover I don't see any analysis of the gain in this PR. I'm globally in favor of doing this development as it is a "easy" gain but it is not a hurry and it should  be properly tested so I propose to postpone the merge until we have someone that has time to do this.

**Comment by adriguir** [comment]
Do I convert it to draft thus? To avoid someone merging it.. Or to another status?

**Comment by rosiereflo** [comment]
I converted it into draft

**Comment by adriguir** [comment]
Thanks

**Pull request closed without merging**